### PR TITLE
[BUGFIX beta] Remove FastBoot monkeypatch

### PR DESCRIPTION
**Pull request by tomdale**
The DOM helper used to require a patched `protocolForURL` to work in Node. This PR:
1. Adds a test to verify attributes are correctly sanitized when rendered via `visit()` in Node
2. Removes the monkeypatch

Note that this PR relies on https://github.com/tildeio/htmlbars/pull/437, which will need to be merged and bumped before this PR's tests will pass.

/cc @chancancode

### CODE REVIEW

File: tests/node/app-boot-test.js [hunk 1]
```diff
@@ -8,123 +8,117 @@ function assertHTMLMatches(assert, actualHTML, expectedHTML) {
 
 appModule("App Boot");
 
-// Because running these tests relies on certain feature flags being
-// enabled, we check to see if those flags are on before registering
-// the tests. Once all feature flags are on by default we can remove
-// this conditional.
-if (appModule.canRunTests) {
```
Comment (chancancode) [inline tests/node/app-boot-test.js]: Do we need to keep this around *in case* it gets backed out?

**Comment by chancancode** [comment]
ProTip™: use https://github.com/emberjs/ember.js/pull/12677/files?w=1 for a better diff :eyes:

File: tests/node/app-boot-test.js (hunk 1 above)
Comment (tomdale) [inline tests/node/app-boot-test.js]: I guess so. I'm not sure of the actual process of when things get removed. /cc @rwjblue

**Comment by tomdale** [comment]
Looks like this is failing on IE. I don't have a VM on my laptop at the moment, though I'm trying to download one on slow Wi-Fi. If anyone with IE handy could run the tests and report the error, I would appreciate it.

**Comment by tomdale** [comment]
Just a sauce issue, thanks @rwjblue for the re-run.

**Comment by tomdale** [comment]
Added `[BUGFIX beta]` tag

**Comment by rwjblue** [comment]
Thank you @tomdale

**Pull request merged**